### PR TITLE
tpl: Fix hugo package name and add godocs

### PR DESCRIPTION
**Pull request by moorereason**

### CODE REVIEW

File: tpl/hugo/init.go [hunk 1]
```diff
@@ -11,7 +11,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package site
+// Package hugo provides template functions for accessing the Site Hugo object.
```
Comment (bep) [inline tpl/hugo/init.go]: What is a "Site Hugo" object?

I'm not sure what the correct term is, but that isn't. And this doesn't really "provide template functions accessing ..."

File: tpl/reflect/init.go [hunk 2]
```diff
@@ -11,6 +11,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Package reflect provides template functions for run-time object reflection.
```
Comment (bep) [inline tpl/reflect/init.go]: We need to do better than that; what functionality does the package provide?

File: tpl/site/init.go [hunk 3]
```diff
@@ -11,6 +11,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Package site provides template functions for accessing the Site object.
```
Comment (bep) [inline tpl/site/init.go]: Does it?

File: tpl/hugo/init.go (hunk 1 above)
Comment (moorereason) [reply tpl/hugo/init.go]: How about `Package hugo provides easy access to Hugo version information.`?

I'm working on generating template namespace docs for the docs repo, and I'm using the package godoc as the namespace description.

File: tpl/site/init.go (hunk 3 above)
Comment (moorereason) [reply tpl/site/init.go]: Does it not?

File: tpl/reflect/init.go (hunk 2 above)
Comment (moorereason) [reply tpl/reflect/init.go]: `Package reflect provides template functions for identifying object types.`?

Offer a better description.

File: tpl/hugo/init.go (hunk 1 above)
Comment (moorereason) [reply tpl/hugo/init.go]: BTW, the docs generation is a one-time event right now.  I don't intend to update the docs site continually with what I'm doing.  Just writing a bash script to do the heaving lifting.

**Pull request merged**